src/main/java/seedu/address/storage/JsonAdaptedGroup.java

```diff
@@ -75,12 +81,20 @@ public JsonAdaptedGroup(Group source) {
      *
      * @throws IllegalValueException if there were any data constraints violated in the adapted person.
      */
-    public Group toModelType() throws IllegalValueException {
+    public Group toModelType(AddressBook addressBook) throws IllegalValueException {
         final ArrayListMap<Person, GroupMemberDetail> modelGroupMembers = new ArrayListMap<>();
         for (Map.Entry<String, JsonAdaptedGroupMemberDetails> entry : groupMembers.entrySet()) {
-            Person key = entry.getValue().toModelType().getPerson();
-            GroupMemberDetail value = entry.getValue().toModelType();
-            modelGroupMembers.put(key, value);
+            String personName = entry.getKey();
+            GroupMemberDetail groupMemberDetail = entry.getValue().toModelType();
```

Comment:
We should also change JsonAdaptedGroupMemberDetail to only store the person's name rather than the Person object